leetcode/iter_2.rs

```diff
@@ -0,0 +1,39 @@
+//Count Matches
+impl Solution {
+    pub fn number_of_matches(mut n: i32) -> i32 {
+        let mut matches = vec![];
+
+        while n > 1 {
+            matches.push(n / 2);
+
+            if n % 2 == 0 {
+                n /= 2;
+            } else {
+                n /= 2;
+                n += 1;
+            }
+        }
+
+        return matches.iter().fold(0, |sum, x| sum + x);
```

Comment:
Since we need the total number only, `matches` as a vector is unnecessary. Try not to use `Vec`.